src/Compiler/AST/Visitor/ExprConverter.cpp

```diff
@@ -262,55 +266,111 @@ void ExprConverter::ConvertExprImageAccessArray(ExprPtr& expr, ArrayExpr* arrayE
             {
                 /* Get buffer type denoter from array indices of array access plus identifier */
                 auto bufferTypeDen = bufferDecl->GetTypeDenoter()->GetSub(arrayExpr);
-
-                if (!arrayExpr->arrayIndices.empty())
+                if (auto baseBufferTypeDen = bufferTypeDen->As<BaseTypeDenoter>())
```

Comment:
I wasn't clear on what `GetSub` method does, but in my tests bufferTypeDen was a `BaseTypeDenoter`. 

The test code was:
```
RWBuffer<float4> gOutputBuf;
Buffer<float4> gInputBuf;

[numthreads(8, 8, 1)]
void main(
	uint3 dispatchThreadId : SV_DispatchThreadID,
	uint threadIndex : SV_GroupIndex)
{
	float4 value = gInputBuf[threadIndex];
	gOutputBuf[threadIndex] = value;
}
```

What would be the proper way to handle this?